### PR TITLE
docs: refresh dev entrypoints in CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by erdii**
Signed-off-by: Josh Gwosdz <jgwosdz@redhat.com>

### Summary

Browsing through this file in our refinement made me realize that the documented entrypoints where slightly outdated and missing an explanation of the IMAGE_REGISTRY override.

### Change Type

Docs/Test

### Check List Before Merging

- [x] This PR passes all pre-commit hook validations.
- [x] This PR is fully tested and regression tests are included.
- [x] Relevant documentation has been updated.
- [x] The commits in this PR follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
      standard.

### Additional Information

Preview link to the changed markdown file: https://github.com/erdii/package-operator/blob/update-developer-documentation/CONTRIBUTING.md

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -42,29 +44,53 @@ hooks if needed. When installed the `pre-commit` hooks will be run on every `git
...
+- package-operator-manager: will pull package images when a `(Cluster)Package` object changes.
+- kubelet/container-runtime: will pull workload images specified in `Pod` objects.
 
-#### Accessing a cluster deployed using dev:deploy
+If you need to access (package) images from `quay.io` you can move the override out of the way by:
+1. Ensuring to fully destroy your current development environment, by running: `./do Dev:Destroy`.
+2. Recreating the development environment with another registry prefix (that will be redirected to localhost:5000, which in turn frees up access to quay.io): `IMAGE_REGISTRY=ctr.package-operator.run/dev ./do Dev:Create`.
+3. Prepend all other cardboard commands with `IMAGE_REGISTRY=ctr.package-operator.run/dev` (or export it via an `.envrc`).
+
+We're thinking of changing this away from quay.io by default.
```
Comment (pbabic-redhat) [inline CONTRIBUTING.md:80]: Should this be here? Seems more like a personal note/todo.

Comment (erdii) [reply CONTRIBUTING.md:80]: > We're thinking of changing this away from quay.io by default.

this line alone?

It's there for trying to invite criticism of this default and to communicate that we're open to chaning it, if one wants to complain about it.
But I can remove it as well.

Comment (pbabic-redhat) [reply CONTRIBUTING.md:80]: Alright, I'm fine either way, no need to change.

**Review by pbabic-redhat — APPROVED** [review]
lgtm

**Pull request merged**